backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/ClusterFactory.java

```diff
@@ -19,51 +19,135 @@
-        }
-        return clusterBuilder.build();
+
+    private Optional<String> username;
+    private Optional<String> password;
+
+    private Optional<String> host;
+    private Optional<Integer> port;
+    private Optional<Collection<CassandraNodeIpAndPort>> servers;
+
+    private Optional<Integer> refreshSchemaIntervalMillis;
+    private boolean forTest;
+
+    private ClusterFactory() {
+        username = Optional.empty();
+        password = Optional.empty();
+
+        host = Optional.empty();
+        port = Optional.empty();
+        servers = Optional.empty();
+
+        refreshSchemaIntervalMillis = Optional.empty();
+        forTest = false;
+    }
+
+    public ClusterFactory username(String username) {
+        this.username = Optional.of(username);
+
+        return this;
+    }
+
+    public ClusterFactory password(String password) {
+        this.password = Optional.of(password);
+
+        return this;
+    }
+
+    public ClusterFactory host(String host) {
+        this.host = Optional.of(host);
+
+        return this;
+    }
+
+    public ClusterFactory port(int port) {
+        this.port = Optional.of(port);
+
+        return this;
     }
 
-    public static Cluster createClusterForClusterWithoutPassWord(List<CassandraNodeIpAndPort> servers) {
-        return createClusterForClusterWithPassWord(servers, null, null, Optional.empty());
+    public ClusterFactory refreshSchemaIntervalMillis(int refreshSchemaIntervalMillis) {
+        this.refreshSchemaIntervalMillis = Optional.of(refreshSchemaIntervalMillis);
+
+        return this;
     }
 
-    public static Cluster createClusterForSingleServerWithPassWord(String ip, int port, String userName, String password) {
-        return createClusterForClusterWithPassWord(ImmutableList.of(new CassandraNodeIpAndPort(ip, port)), userName, password, Optional.empty());
+    public ClusterFactory servers(CassandraNodeIpAndPort ...servers) {
+        this.servers = Optional.of(ImmutableList.copyOf(servers));
+
+        return this;
     }
 
-    public static Cluster createClusterForSingleServerWithoutPassWord(String ip, int port) {
-        return createClusterForClusterWithPassWord(ImmutableList.of(new CassandraNodeIpAndPort(ip, port)), null, null, Optional.empty());
+    public ClusterFactory servers(Collection<CassandraNodeIpAndPort> servers) {
+        this.servers = Optional.of(servers);
+
+        return this;
     }
 
-    public static Cluster createTestingCluster(String ip, int port) {
-        return createClusterForClusterWithPassWord(ImmutableList.of(new CassandraNodeIpAndPort(ip, port)), null, null, Optional.of(0));
+    public ClusterFactory forTest() {
+        forTest = true;
+
+        return this;
     }
 
-    public static Cluster createDefaultSession() {
-        return createClusterForSingleServerWithoutPassWord(DEFAULT_CLUSTER_IP, CassandraNodeIpAndPort.DEFAULT_CASSANDRA_PORT);
+
+    public Cluster build() {
```

Comment:
Should I introduce unit test for all this logic ? 
If yes I think I will need something such as PowerMockito to mock Cluster.Builder. Is there already something available for that ?